lib/vdc/GridHelper.cpp

```diff
@@ -276,12 +278,11 @@ LayeredGrid *GridHelper::_make_grid_layered(
 
 	// Get horizontal dimensions
 	//
-	vector <double> hminu, hmaxu;
-	for (int i = 0; i<2; i++) { 
-		float *coords = blkvec[i+1];
-		hminu.push_back(coords[0]);
-		hmaxu.push_back(coords[dims[i]-1]);
-	}
+	vector <double> xcoords;
+	for (int i=0; i<dims[0]; i++) xcoords.push_back(blkvec[1][i]);
+
+	vector <double> ycoords;
+	for (int i=0; i<dims[1]; i++) ycoords.push_back(blkvec[2][i]);
```

Comment:
Depending on if this piece of code is performance critical, you could call the reserve() method of the vector before pushing back a bunch of values.

Reply:
Yup. Thanks. But the constructors in general were not designed to be performant. If the need ever arrises lots of changes will be required :-)